src/runtime/refiner.ts

```diff
@@ -627,13 +667,13 @@ export class Segment {
     let right: Boundary;
     if (a.from.val === b.from.val) {
       left = {...a.from};
-      left.kind = a.from.kind === b.from.kind ? a.from.kind : 'open';
+      left.isOpen = a.from.isOpen === b.from.isOpen ? a.from.isOpen : true;
     } else {
       left = a.from.val > b.from.val ? {...a.from} : {...b.from};
     }
     if (a.to.val === b.to.val) {
       right = {...a.to};
-      right.kind = a.to.kind === b.to.kind ? a.to.kind : 'open';
+      right.isOpen = a.to.isOpen === b.to.isOpen ? a.to.isOpen : true;
```

Comment:
This seems to be the same as `a.to.isOpen || b.to.isOpen`. Can you double check for me?